### ISSUE TITLE
Use `indoc!()` for multiline strings 

### ISSUE THREAD

**Issue by umucorn** [denoland/deno_lint#597]
https://docs.rs/indoc/1.0.3/indoc/ crate provides functionality for using nested multiline strings. We use multiline strings for documentation and tests. But they look totally out of order in our code.

It improves code readability. 

Example:

```rust
#[test]
fn explicit_function_return_type_invalid() {
  assert_lint_err::<ExplicitFunctionReturnType>("function foo() { }", 0);
  assert_lint_err_on_line_n::<ExplicitFunctionReturnType>(
    indoc!{r#"
      function a() {
        function b() {}
      }
    "#},
    vec![(2, 0), (3, 2)],
  );
}
```

would be equivalent to:

```rust
#[test]
  fn explicit_function_return_type_invalid() {
    assert_lint_err::<ExplicitFunctionReturnType>("function foo() { }", 0);
    assert_lint_err_on_line_n::<ExplicitFunctionReturnType>(
      r#"
function a() {
  function b() {}
}
      "#,
      vec![(2, 0), (3, 2)],
    );
  }
```

**Comment by henryboisdequin** [comment]
I would like to work on this issue.

**Comment by henryboisdequin** [comment]
I found that using the macro `indoc` actually breaks things. In your code example, the test fails.

```
thread 'rules::explicit_function_return_type::tests::explicit_function_return_type_invalid' panicked at 'expect diagnostics explicit-function-return-type at 1:0 to be explicit-function-return-type at 2:0
```

**Comment by umucorn** [comment]
> I found that using the macro `indoc` actually breaks things. In your code example, the test fails.
> 
> ```
> thread 'rules::explicit_function_return_type::tests::explicit_function_return_type_invalid' panicked at 'expect diagnostics explicit-function-return-type at 1:0 to be explicit-function-return-type at 2:0
> ```

I didn't test my example. But it's interesting now. Or I didn't get how this macro works.

**Comment by kitsonk** [comment]
Let's not fix something that works with something that doesn't.

Respectfully closing.
